src/graphql/data/TokenPrice.ts

```diff
@@ -1,84 +1,60 @@
 import graphql from 'babel-plugin-relay/macro'
-import { useEffect, useMemo, useState } from 'react'
-import { fetchQuery } from 'react-relay'
+import { useEffect, useMemo } from 'react'
+import { loadQuery, PreloadedQuery, usePreloadedQuery, useRelayEnvironment } from 'react-relay'
 
 import { Chain, TokenPriceQuery } from './__generated__/TokenPriceQuery.graphql'
-import environment from './RelayEnvironment'
-import { TimePeriod } from './util'
+import { TimePeriod, toHistoryDuration } from './util'
 
-const tokenPriceQuery = graphql`
-  query TokenPriceQuery($contract: ContractInput!) {
+export const tokenPriceQuery = graphql`
+  query TokenPriceQuery($contract: ContractInput!, $duration: HistoryDuration!) {
     tokens(contracts: [$contract]) {
-      market(currency: USD) {
-        priceHistory1H: priceHistory(duration: HOUR) {
-          timestamp
-          value
+      market(currency: USD) @required(action: LOG) {
+        price {
+          value @required(action: LOG)
         }
-        priceHistory1D: priceHistory(duration: DAY) {
-          timestamp
-          value
-        }
-        priceHistory1W: priceHistory(duration: WEEK) {
-          timestamp
-          value
-        }
-        priceHistory1M: priceHistory(duration: MONTH) {
-          timestamp
-          value
-        }
-        priceHistory1Y: priceHistory(duration: YEAR) {
-          timestamp
-          value
+        priceHistory(duration: $duration) {
+          timestamp @required(action: LOG)
+          value @required(action: LOG)
         }
       }
     }
   }
 `
-
-export type PricePoint = { timestamp: number; value: number }
+export type PricePoint = NonNullable<
```

Comment:
I find this harder to read, and would prefer if it were explicit. If the types mismatch it should still be caught by the checker. Could you revert back to

    export type PricePoint = { timestamp: number; value: number }